src/Lidarr.Api.V1/Artist/ArtistModule.cs

```diff
@@ -245,17 +245,17 @@ public void Handle(TrackFileDeletedEvent message)
         {
             if (message.Reason == DeleteMediaFileReason.Upgrade) return;
 
-            BroadcastResourceChange(ModelAction.Updated, message.TrackFile.ArtistId);
+            BroadcastResourceChange(ModelAction.Updated, message.TrackFile.Artist.Value.ToResource());
         }
 
         public void Handle(ArtistUpdatedEvent message)
         {
-            BroadcastResourceChange(ModelAction.Updated, message.Artist.Id);
+            BroadcastResourceChange(ModelAction.Updated, message.Artist.ToResource());
```

Comment:
This broadcast resource changes break updates to UI. Changing back to Id causes updates to come thru as expected. 

Ex. Import artist and sit on artist index page. At end of import final stats don't get updated correctly, However reverting to using function with `int Id` overlord causes resource to be retrieved fully and then sent to the UI.

This is because simply passing the `artist.toResource()` doesn't populate the entire resource (ex. statistics property  is left as `new ArtistStatisticsResource()`.) When using the `BroadcastResourceChange(ModelAction.Updated, message.Artist.Id)` overlord the function grabs the resource by using `GetResourceById` which in the artist case triggers the `GetArtist` function in ArtistModule.cs to return the resource. This function fully populates the resource (adds statistics, covers, etc.) and then broadcasts to the UI. Best to leave as is and use the Id.

Reply:
Good catch